### ISSUE TITLE
Sso on custom home server with android does not work

### ISSUE THREAD

**Issue by mqus** [syphon-org/syphon#335]
When I try to log in via sso on my homeserver, the auth form opens in the browser and I can login. When I'm finished, the browser tries to redirect to syphon://... But this just opens the syphon app from a fresh state(even with the disclaimer popup) and not logging my in.

I can try to connect adb If you need more info :)

**Comment by maxkratz** [comment]
A similar behavior shows on my homeserver when I try to login via the iOS app. The Webbrowser redirects me to `syphon://...` and the app shows me the SSO login page again. (Attached screenshot shows the app status *after* logging in via the browser!)

![21-10-03 13-15-20 0865](https://user-images.githubusercontent.com/20956405/135751262-a0fbd323-07f3-49cf-8d73-b3e77b5f0dce.png)

**Comment by frcl** [comment]
I have the same Problem with a homeserver where I try to sign in via OpenID. My syphon version is 0.2.0.

**Comment by ser** [comment]
It still does not work. Syphon says that he is unable to get "well known" from my synapse, which is obviously something wrong. Other clients as fluffy or element are logging in properly

**Comment by frcl** [comment]
> Syphon says that he is unable to get "well known" from my synapse, which is obviously something wrong.

This is true but, it is probably not connected to the issue described here. The "faild well known check" message appears for homeservers which to not provide a /.well-known/matrix/client which is a json file the can be used to redirect to the actual server address. [Accourding to the matrix spec](https://spec.matrix.org/v1.1/client-server-api/#getwell-knownmatrixclient) this is optional and if the server returns 404 the client should ignore it. This is not [how syphon does it](https://github.com/syphon-org/syphon/blob/58555d1a4059f165fc72c6d4c7507914f722204d/lib/global/libs/matrix/auth.dart#L315), but the original issue is about being unable to login without error even when the "well known check" passes.

**Comment by ser** [comment]
It's connected in some way as because of "well known" syphon does not show me SAML login option at all.

**Comment by menturion** [comment]
I am having a similar problem with a homeserver running without a proxy and TLS-enabled listeners on port `8448`, i.e. ...

`static const  homeserverDefault = 'example.com:8448';`

Does Syphon have any (SSO) issues with such a setup?

**Comment by frcl** [comment]
I still have the issue with version 2.8, but I have made an important observation: The problem is browser dependent.

With DuckDuckGo browser as default on my phone I experience the issue, but when I use a different Browser (e.g. EinkBro from F-Droid) the login process works without issues.

With DuckDuckGo it seems like the `syphon://…` link is not opened in the Syphon app window, but in the DuckDuckGo app window instead (see screenshort). I don't know if this is connected to the issue, but in EinkBro it is different.

@mqus to confirm, what is your default browser?

![Screenshot_System_UI_20220208-175102](https://user-images.githubusercontent.com/7346166/153038727-6ed67371-456c-448f-b870-c4508c9734da.png)

**Comment by mqus** [comment]
Its firefox on android, but the old one

**Comment by mqus** [comment]
Yeah, quickndirty: I get the .well-known error on the current version and after that the sso-login-button disappeared in syphon. I'll ask the admins of my instance. I can pn you the domain if you need it.
